src/extra/JsonSchemaRenderer/JsonSchemaRendererPlayground/util.ts

```diff
@@ -0,0 +1,123 @@
+import { UiSchema, JSONSchema } from '../types';
+import {
+	first,
+	forEach,
+	get,
+	without,
+	uniq,
+	castArray,
+	keys,
+	cloneDeep,
+} from 'lodash';
+import widgets from '../widgets';
+import { getWidget, getType, JsonSchemaRendererProps } from '../index';
+import { WidgetWrapperUiOptions, getObjectPropertyNames } from '../widgets';
+import { Value } from '../types';
+
+type UiSchemaMetaSchema = {
+	title?: string;
+	description?: string;
+	type: string | string[];
+	properties: {
+		[key: string]: any;
+	};
+};
+
+const baseMetaSchema: UiSchemaMetaSchema = {
+	title: 'JsonSchemaRenderer UI Schema',
+	type: ['object', 'null'],
+	properties: {
+		'ui:title': {
+			type: ['string', 'null'],
+		},
+		'ui:description': {
+			type: ['string', 'null'],
+		},
+		'ui:value': {
+			type: ['string', 'object', 'null'],
+		},
+		'ui:options': {
+			type: 'object',
+		},
+	},
+};
+
+export const generateUiSchemaMetaSchema = ({
```

Comment:
Can we break down the function a bit? It is quite nested and long, so it's difficult to follow what's going on